src/main/kotlin/com/autonomousapps/subplugin/ProjectPlugin.kt

```diff
@@ -58,8 +58,6 @@ internal class ProjectPlugin(private val project: Project) {
     }
   }
 
-  private lateinit var inMemoryCacheProvider: Provider<InMemoryCache>
```

Comment:
Removed to not register too early (i.e. before all builds got the chance to apply the plugin).

Also, this additional state is not necessary. The `registerIfAbsent` of Gradle keeps care of holding the already registered service and should be optimized for being called many times.